transports/quic/tests/smoke.rs

```diff
@@ -183,6 +183,98 @@ fn concurrent_connections_and_streams_tokio() {
         .quickcheck(prop::<quic::tokio::Provider> as fn(_, _) -> _);
 }
 
+#[cfg(feature = "tokio")]
+#[tokio::test]
+async fn draft_29_support() {
+    use std::task::Poll;
+
+    use futures::{future::poll_fn, select};
+    use libp2p::TransportError;
+
+    let _ = env_logger::try_init();
+
+    let (_, mut a_transport) =
+        create_transport::<quic::tokio::Provider>(|cfg| cfg.support_draft_29 = true);
+    let (_, mut b_transport) =
+        create_transport::<quic::tokio::Provider>(|cfg| cfg.support_draft_29 = true);
+
+    // If a server supports draft-29 all its QUIC addresses can be dialed on draft-29 or version-1
+    let a_quic_addr = start_listening(&mut a_transport, "/ip4/127.0.0.1/udp/0/quic").await;
+    let a_quic_mapped_addr = a_quic_addr
+        .clone()
+        .into_iter()
+        .map(|p| {
+            if matches!(p, Protocol::QuicV1) {
+                Protocol::Quic
+            } else {
+                p
+            }
+        })
+        .collect();
```

Comment:
This snippet is repeated 3 times in the test. Can we make a little `swap_protocol` utility so these are one-liners?

Reply:
Wdyt of 4cd494d?